verifier/testreport.py

```diff
@@ -748,24 +758,28 @@ def characterize_datetime_tests(self, test_list, results):
     def add_to_results_by_key(self, label, results, input_data, test, key_list):
         if input_data:
             for key in key_list:
-                if input_data.get(key, None):  # For collation results
-                    value = input_data.get(key, None)
-                    if key == 'input_list':
-                        if 'input_size' not in results:
-                            results['input_size'] = {}
-                        else:
-                            results['input_size'].add(len(value))
-                    if key == 'rules':
-                        value = 'RULE'  # A special case to avoid over-characterization
-                    if key not in results:
-                        results[key] = {}
-                    try:
-                        if not results[key].get(value, None):
-                            results[key][value] = set()
-                        results[key][value].add(label)
-                    except TypeError as err:
-                        # value may not be hashable. This should be skipped
-                        pass
+                try:
+                    if input_data.get(key, None):  # For collation results
+                        value = input_data.get(key, None)
+                        if key == 'input_list':
+                            if 'input_size' not in results:
+                                results['input_size'] = {}
+                            else:
+                                results['input_size'].add(len(value))
+                        if key == 'rules':
+                            value = 'RULE'  # A special case to avoid over-characterization
+                        if key not in results:
+                            results[key] = {}
+                        try:
+                            if not results[key].get(value, None):
+                                results[key][value] = set()
+                            results[key][value].add(label)
```

Comment:
See if you can assure that wherever the initialization of `input_data` occurs uses `defaultdict`, and then simplify the code here.

Reply:
Trying this now.